lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -116,9 +99,64 @@ def pg_connection_args
         }.freeze
       end
 
-      def clear_caches
-        @sources = nil
-        @collector_deployments = nil
+
+      ### Orchestrator Stuff
+      def collector_definitions
+        @collector_definitions ||= begin
+          require 'yaml'
+          YAML.load_file(@collector_definitions_file)
+        end
+      end
+
+      def object_manager
+        @object_manager ||= ObjectManager.new
+      end
+
+
+      ### Openshift stuff
+      def collectors_from_openshift
+        object_manager.get_deployment_configs("topological-inventory/collector=true").collect { |i| i.metadata.labels["topological-inventory/collector_digest"] }
+      end
+
+      def create_openshift_objects_for_source(digest, source)
+        puts "Creating objects for source #{source["source_id"]} with digest #{digest}"
+        object_manager.create_secret(collector_deployment_secret_name_for_source(source), source["secret"])
+        object_manager.create_deployment_config(collector_deployment_name_for_source(source)) do |d|
+          d[:metadata][:labels]["topological-inventory/collector_digest"] = digest
+          d[:metadata][:labels]["topological-inventory/collector"] = "true"
+          d[:spec][:replicas] = 1
+          container = d[:spec][:template][:spec][:containers].first
+          container[:env] = collector_container_environment(source)
+          container[:image] = source["image"]
+        end
+      end
+
+      def remove_openshift_objects_for_source(digest)
+        return unless digest
+        deployment = object_manager.get_deployment_configs("topological-inventory/collector_digest=#{digest}").detect { |i| i.metadata.labels["topological-inventory/collector"] == "true" }
+        return unless deployment
+        puts "Removing objects for deployment #{deployment.metadata.name}"
+        object_manager.delete_deployment_config(deployment.metadata.name)
+        object_manager.delete_secret("#{deployment.metadata.name}-secrets")
+      end
+
+      def collector_deployment_name_for_source(source)
+        "topological-inventory-collector-source-#{source["source_id"]}"
+      end
+
+      def collector_deployment_secret_name_for_source(source)
+        "#{collector_deployment_name_for_source(source)}-secrets"
+      end
+
+      def collector_container_environment(source)
+        secret_name = "#{collector_deployment_name_for_source(source)}-secrets"
+        [
+          {:name => "AUTH_PASSWORD", :valueFrom => {:secretKeyRef => {:name => secret_name, :key => "password"}}},
+          {:name => "AUTH_USERNAME", :valueFrom => {:secretKeyRef => {:name => secret_name, :key => "username"}}},
+          {:name => "INGRESS_API", :value => ENV["TOPOLOGICAL_INVENTORY_INGRESS_API_PORT"].sub("tcp://", "http://")},
```

Comment:
I think I'd prefer this be composed as:
```ruby
"http://#{ENV["TOPOLOGICAL_INVENTORY_INGRESS_API_SERVICE_HOST"]}:#{ENV["TOPOLOGICAL_INVENTORY_INGRESS_API_SERVICE_PORT"]}"
```

Alternatively we could pass it into the orchestrator as a single ENV variable like we are doing currently for collectors: https://github.com/ManageIQ/topological_inventory-deploy/blob/e8d28a52e30fac40ad03c9ee3ea74abb4cb8f1e1/amazon-collector/deploy_template.yaml#L43